argilla-server/src/argilla_server/api/schemas/v1/records.py

```diff
@@ -82,7 +82,7 @@ class Config:
 
 
 class RecordCreate(BaseModel):
-    fields: Dict[str, Union[StrictStr, None]]
+    fields: Dict[str, Union[StrictStr, None, List[Dict[StrictStr, StrictStr]]]]
```

Comment:
I believe we should define a Pydantic model specific for the chat field.

So we define the `fields` attribute with something similar to:

```suggestion
    fields: Dict[str, Union[StrictStr, List[RecordChatField], None]]
```

And with `RecordChatField` being:

```python
class RecordChatField(BaseModel):
    role: str
    content: str
```

You can add validations to `RecordChatField`, define a `RecordChatFieldCreate` and `RecordChatFieldUpdate` if necessary, etc.

If necessary an alternative to this is define a `RecordChatFieldValue` as a list of `RecordChatField`'s with validations:

```suggestion
    fields: Dict[str, Union[StrictStr, RecordChatFieldValue, None]]
```

And with `RecordChatFieldValue` being:

```python
RecordChatFieldValue = Annotated[conlist(item_type=RecordChatField, min_items=1), Field(..., description="...")] 
```

Reply:
I agree that a Pydantic model would be ideal here. 

However, we want to allow the object to have more keys/values than just 'role' and 'content' because these structures aren't fixed. Let's discuss this.